documentation/homework/week4/src/components/index.js

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+
+import { inject, observer } from 'mobx-react'
+
+import AssignItem from './AssignItem'
+import ViewRender from './ViewRender'
+import EditRender from './EditRender'
+import { locals } from '../utils'
+
+@inject('InitialStore')
+@observer
+export default class extends Component {
+  componentDidUpdate = (nextProps) => locals.save = [...nextProps.InitialStore.items]
+  render() {
+    const { items, removeItem, toggle_edit, toggle_checkbox, submit_edit, item_edit_state, onInputEdit } = this.props.InitialStore
```

Comment:
So you are connecting your `Index` component to the store, but then passing a lot of things as props instead of connecting the child components directly. Both options work, but I wonder if you had a reason for this?

The main point of using a Provider is that we can connect child components anywhere in the tree to the data they need, even if it's just one field from the store.

Reply:
Actully my reason that I hate the code that has too many imports inside it and some times I feel it very messy, Well you might say we have the `@inject` and the `@observer` too.
But It's just a one level and there is not that many `props` I am passing for each one.

And i don't know i just felt more comfortable working with the child component, because the hall almost work in two places the stores and the last child component.